framework/src/engine/bft/method.ts

```diff
@@ -210,6 +195,14 @@ export class BFTMethod {
 			validatorsHash,
 		};
 
+		const votesStore = stateStore.getStore(MODULE_STORE_PREFIX_BFT, STORE_PREFIX_BFT_VOTES);
+		const bftVotes = await votesStore.getWithSchema<BFTVotes>(EMPTY_KEY, bftVotesSchema);
+		// This assumes bftVotes.blockBFTInfos will contain currently executing block
+		const nextHeight =
+			(bftVotes.blockBFTInfos.length > 0
+				? bftVotes.blockBFTInfos[0].height
```

Comment:
I think this is different in specification, as per the LIP it should be `bftVotes.blockBFTInfos[0].height + 1`

Reply:
I am not sure how. I see the existing implementation (dev branch) also uses `bftVotes.blockBFTInfos[0].height + 1` for `nextHeight` similar to the LIP.

Reply:
ah that's true, the current height is `bftVotes.blockBFTInfos[0].height` and `nextHeight` is `+1`

Reply:
Re-checking the code, the next height is
```
const nextHeight =
			(bftVotes.blockBFTInfos.length > 0
				? bftVotes.blockBFTInfos[0].height
				: bftVotes.maxHeightPrevoted) + 1;
```
it is adding `1` at the end?

Reply:
ah, I see now there is a parenthesis in the end. my bad, please disregard this and other comment about the test.